vulkan_helpers/helper_functions.cpp

```diff
@@ -1142,10 +1140,12 @@ VkSwapchainKHR CreateDefaultSwapchain(
     }
 
     uint32_t minSwapchains =
-        std::max(surface_caps.minImageCount + 1, min_image_count + 1);
+        min_image_count ? min_image_count : surface_caps.minImageCount + 1;
```

Comment:
I see where this fixes the issue where min_image_count is less than the cap min, I suppose I don't understand why the default min swapchain is the cap min + 1 in the first place and you dont't just do 
minSwapchains = std::max(min_image_count, surface_cap.minImageCount)

Reply:
This was a hold-over from an ancient sample. I didn't really want to change that behavior.
The behavior is mostly because MOST swapchains report minImageCount of 1, which means you wont be able to keep 2 frames in flight at a time.
I will update this logic.